src/ledger/LedgerManager.h

```diff
@@ -151,6 +151,13 @@ class LedgerManager
     virtual void startCatchup(CatchupConfiguration configuration,
                               bool manualCatchup) = 0;
 
+    // Queue a batch of LedgerEntries to send to the BucketManager on the
+    // next ledger, instead of the entries generated by the LedgerTxn.
+    virtual void setNextLedgerEntryBatchForBucketTesting(
```

Comment:
would have it been possible to define this in a derived testing class instead? It seems all you're trying to do is change the behavior of `LedgerManagerImpl::ledgerClosed` so having a custom `LedgerManagerImplForTesting::ledgerClosed` may be the better choice (and can be moved to test file)

Reply:
Fixed.